.github/workflows/argilla-server.yml

```diff
@@ -0,0 +1,192 @@
+name: Build Argilla server package
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+
+  pull_request:
+    paths:
+      - "argilla-server/**"
+    types:
+      - opened
+      - edited
+      - reopened
+      - synchronize
+      - ready_for_review
+
+  release:
+    types:
+      - "published"
+
+jobs:
+
+  build:
+    name: Build `argilla-server` package
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash -l {0}
+        working-directory: argilla-server
+
+    services:
+      elasticsearch:
+        image: docker.elastic.co/elasticsearch/elasticsearch:8.8.2
+        ports:
+          - 9200:9200
+        env:
+          discovery.type: single-node
+          xpack.security.enabled: false
+
+      postgres:
+        image: postgres:14
+        env:
+          POSTGRES_HOST: localhost
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: argilla
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    env:
+      ARGILLA_ENABLE_TELEMETRY: 0
+
+    steps:
+      - name: Checkout Code 🛎
+        uses: actions/checkout@v4
+
+      - name: Setup PDM
+        uses: pdm-project/setup-pdm@v4
+        with:
+          python-version-file: argilla-server/pyproject.toml
+          cache-dependency-path: argilla-server/pdm.lock
+          cache: true
+
+      - name: Install dependencies
+        run: pdm install
+
+      - name: Run tests 📈
```

Comment:
We're running tests only with PostgreSQL?

Reply:
Yes. I will include a separate step to run tests with SQLite (and other python engines) later.